optimum/bettertransformer/transformation.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2022 The HuggingFace and Meta Team.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from copy import deepcopy
+
+import torch
+
+from .models import is_module_fast
+from .utils import check_if_pytorch_greater_112, is_accelerate_available
+
+
+if is_accelerate_available():
+    from accelerate import dispatch_model, infer_auto_device_map
+    from accelerate.hooks import remove_hook_from_module
+    from accelerate.utils import named_module_tensors, set_module_tensor_to_device
+
+
+def init_accelerate_hook(module):
+    r"""
+    This function initializes `accelerate` hooks by setting the
+    parameters of the module on the `cpu` if there is any offloading involved
+
+    Args:
+        `module` (`torch.nn.Module`, **required**):
+            The input module to initialize
+    Returns:
+        The initialized module
+    """
+    for name, child in module.named_children():
+        if hasattr(child, "_hf_hook"):
+            if child._hf_hook.weights_map:
+                hook = child._hf_hook
+                if child._hf_hook.offload:
+                    for name, _ in named_module_tensors(
+                        child, include_buffers=hook.offload_buffers, recurse=hook.place_submodules
+                    ):
+                        set_module_tensor_to_device(child, name, "cpu", value=hook.weights_map[name])
+        init_accelerate_hook(child)
+    return module
+
+
+def remove_hooks_recursive(model):
+    r"""
+    Utility function to clean the hooks of the model
+    this is useful when overriding the parent model (i.e. when `keep_original_model`=`True`)
+
+    Args:
+        `module` (`torch.nn.Module`, **required**):
+            The input module to initialize
+    """
+    for _, child in model.named_children():
+        remove_hook_from_module(child)
+        if len(list(child.children())) > 1:
+            remove_hooks_recursive(child)
```

Comment:
Could be upstramed to Accelerate (we can add a `recurse` keyword to `remove_hook_from_module`.

Reply:
Nice! Will open a PR soon ;)

Reply:
Closing as https://github.com/huggingface/accelerate/pull/812#event-7724004483 being merged and 87efe81c90b256bd15fefb4f23c9d2102286f27d